### PR TITLE
webhook sink

### PR DESCRIPTION
**Pull request by arikalon1**

### CODE REVIEW

File: src/robusta/core/sinks/webhook/webhook_sink.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                """
+            )
+            while len(message) + len(wrapped) >= self.size_limit:
+                break
+            message += wrapped
+
+        requests.post(self.url, data=message)
+
+    @classmethod
+    def __to_cleat_text(cls, markdown_text: str) -> str:
```
Comment (aantn) [inline src/robusta/core/sinks/webhook/webhook_sink.py:46]: typo - should be *clear*

**Comment by aantn** [comment]
I like that we're doing this a generic webhook and not something specific to NFTY.

@RoiGlinik leaving a real code review to you. I didn't read most of the code.

File: src/robusta/core/sinks/webhook/webhook_sink.py (hunk 1 above)
Comment (arikalon1) [reply src/robusta/core/sinks/webhook/webhook_sink.py]: fixed, thx

File: src/robusta/core/sinks/webhook/webhook_sink.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            for block in enrichment.blocks:
+                message_lines.extend(self.__to_unformatted_text(block))
+
+        for line in [line for line in message_lines if line]:
+            wrapped = textwrap.dedent(
+                f"""
+                {line}
+                """
+            )
+            while len(message) + len(wrapped) >= self.size_limit:
```
Comment (RoiGlinik) [inline src/robusta/core/sinks/webhook/webhook_sink.py:39]: I think an if will be less confusing here

**Review by RoiGlinik — APPROVED** [review]
I had a small comment , Otherwise LG

File: src/robusta/core/sinks/webhook/webhook_sink.py (hunk 2 above)
Comment (arikalon1) [reply src/robusta/core/sinks/webhook/webhook_sink.py:39]: Will breaking it to 2 lines make it better?
non_empty_lines = [line for line in message_lines if line]
for line in non_empty_lines: ...

Comment (arikalon1) [reply src/robusta/core/sinks/webhook/webhook_sink.py:39]: Good catch. Fixed

**Comment by stephanGarland** [comment]
Great timing; I have a "work on what you'd like" day tomorrow, so I'll be installing this and playing around with it then.

**Comment by arikalon1** [comment]
great. Make sure you use the latest release, 0.8.21( will be released soon)
We'd love to hear any feedback you might have

**Pull request merged**